### PR TITLE
MinPayment null is not usefully distinct from nil

### PR DESCRIPTION
**Pull request by j16r**
We keep a pointer Link in a bunch of structs, but a pointer on a struct takes up more space and nil is not usefully distinct from a 0 value in this case.

So just remove the pointer usage for assets.Link and no more null pointer exceptions!

### CODE REVIEW

File: core/services/runs.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ func MeetsMinimumPayment(
 	expectedMinJobPayment *assets.Link,
 	actualRunPayment *assets.Link) bool {
 	// input.Payment is always present for runs triggered by ethlogs
-	if actualRunPayment == nil || expectedMinJobPayment == nil || expectedMinJobPayment.IsZero() {
+	if actualRunPayment == nil || expectedMinJobPayment.IsZero() {
```
Comment (dimroc) [inline core/services/runs.go]: 🎉

File: core/services/runs.go [hunk 2]
```diff
@@ -18,7 +18,7 @@ func MeetsMinimumPayment(
 	expectedMinJobPayment *assets.Link,
```
Comment (dimroc) [inline core/services/runs.go]: Suggestion: Should we change this from needing a `*assets.Link` to just requiring a value, to be consistent? The comment below checking for `== nil` is still technically needed, unless we can guarantee a nil is never passed in.

File: core/services/run_manager.go [hunk 3]
```diff
@@ -89,7 +89,7 @@ func newRun(
 	run.CreationHeight = models.NewBig(currentHeight)
 	run.ObservedHeight = models.NewBig(currentHeight)
 
-	if !MeetsMinimumPayment(job.MinPayment, payment) {
+	if !MeetsMinimumPayment(&job.MinPayment, payment) {
```
Comment (dimroc) [inline core/services/run_manager.go]: In line with the previous comment, should we just pass the value?

File: core/services/runs.go (hunk 2 above)
Comment (j16r) [reply core/services/runs.go]: Yeah I'll see if this one can be changed as well.

File: core/services/run_manager.go (hunk 3 above)
Comment (j16r) [reply core/services/run_manager.go]: I'm sort of following the convention of passing large values by pointer here.

File: core/services/runs.go (hunk 2 above)
Comment (j16r) [reply core/services/runs.go]: This is really the same one as below, I'd prefer to pass large structures in as pointers as this is the generally accepted go convention.

File: core/services/run_manager.go (hunk 3 above)
Comment (dimroc) [reply core/services/run_manager.go]: Ok, up to you. `MeetsMinimumPayment` is only called twice, always passing `job.MinPayment` and it's entirely inside this file. In the fight against nil reference exceptions, I was shooting for a win.

**Pull request merged**